app/src/main/java/com/github/amusinamaria/ui/MainActivity.kt

```diff
@@ -2,15 +2,21 @@ package com.github.amusinamaria.ui
 
 import android.os.Bundle
 import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentTransaction.TRANSIT_FRAGMENT_FADE
 import com.github.amusinamaria.R
+import com.github.amusinamaria.databinding.ActivityMainBinding
+import com.github.amusinamaria.repository.MovieCard
 import com.github.amusinamaria.ui.details.DetailsFragment
 import com.github.amusinamaria.ui.list.MoviesListFragment
 
-class MainActivity : AppCompatActivity(), MoviesListFragment.MovieClickListener {
+class MainActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMainBinding
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
+        binding = ActivityMainBinding.inflate(layoutInflater)
+        setContentView(binding.root)
 
         if (savedInstanceState == null) {
```

Comment:
Use savedInstanceState ?:  supportFragmentManager.beginTransaction()
This is Kotlin style

Reply:
Done